packages/expo-modules-core/android/src/main/java/expo/modules/kotlin/AppContext.kt

```diff
@@ -58,30 +50,16 @@ import java.lang.ref.WeakReference
 class AppContext(
   modulesProvider: ModulesProvider,
   val legacyModuleRegistry: expo.modules.core.ModuleRegistry,
-  private val reactContextHolder: WeakReference<ReactApplicationContext>
+  reactContextHolder: WeakReference<ReactApplicationContext>
 ) : CurrentActivityProvider {
-  val registry = ModuleRegistry(WeakReference(this))
-  private val reactLifecycleDelegate = ReactLifecycleDelegate(this)
-
-  private var hostWasDestroyed = false
-
-  // We postpone creating the `JSIContext` to not load so files in unit tests.
-  internal lateinit var jsiInterop: JSIContext
 
-  /**
-   * The core module that defines the `expo` object in the global scope of the JS runtime.
-   *
-   * Note: in current implementation this module won't receive any events.
-   */
-  internal val coreModule = run {
-    val module = CoreModule()
-    module._appContext = this
-    ModuleHolder(module)
-  }
+  // The main context used in the app.
+  // Modules attached to this context will be available on the main js context.
+  val hostingContext = RuntimeContext(this, reactContextHolder)
```

Comment:
```suggestion
  val hostingRuntimeContext = RuntimeContext(this, reactContextHolder)
```

it's unclear what's the context here. maybe specify as `hostingRuntimeContext` would be helpful.